File: dynabuffers-python/dynabuffers/__test__/usecase/Schema01Test.py
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema01Test(unittest.TestCase):

    def test_parse(self):
        engine = Dynabuffers.parse(FileStream(__file__ + "/../schema01.dbs"))
        map = engine.deserialize(engine.serialize({"message": {"text": "abcd", "type": ["DEP", "NER"]}}))

        self.assertEqual(map, {"message": {"text": "abcd", "type": ["DEP", "NER"]}})


if __name__ == "__main__":
    unittest.main()

File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema01Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import org.junit.jupiter.api.Test

class Schema01Test : AbstractDynabuffersTest() {

    @Test
    fun testParse() {
        val engine = Dynabuffers.parse(Schema01Test::class.java.getResourceAsStream("/schema01.dbs"))
        val output = mapOf("message" to mapOf("text" to "abcd", "type" to arrayOf("DEP", "NER"), ":type" to 0))
        assertMap(engine, mapOf("message" to mapOf("text" to "abcd", "type" to arrayOf("DEP", "NER"))), output)
    }
}
